mevshare/builders.go

```diff
@@ -0,0 +1,238 @@
+package mevshare
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ybbus/jsonrpc/v3"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+type BuilderAPI uint8
+
+const (
+	BuilderAPIRefRecipient BuilderAPI = iota
+	BuilderAPIMevShareBeta1
+)
+
+var ErrInvalidBuilder = errors.New("invalid builder specification")
+
+type BuildersConfig struct {
+	Builders []struct {
+		Name     string `yaml:"name"`
+		URL      string `yaml:"url"`
+		API      string `yaml:"api"`
+		Internal bool   `yaml:"internal"`
+		Disabled bool   `yaml:"disabled"`
+	} `yaml:"builders"`
+}
+
+// LoadBuilderConfig parses a builder config from a file
+func LoadBuilderConfig(file string) (BuildersBackend, error) {
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return BuildersBackend{}, err
+	}
+
+	var config BuildersConfig
+	err = yaml.Unmarshal(data, &config)
+	if err != nil {
+		return BuildersBackend{}, err
+	}
+
+	externalBuilders := make([]JSONRPCBuilderBackend, 0)
+	internalBuilders := make([]JSONRPCBuilderBackend, 0)
+	for _, builder := range config.Builders {
+		if builder.Disabled {
+			continue
+		}
+
+		var api BuilderAPI
+		switch builder.API {
+		case "refund-recipient":
```

Comment:
suggestion: You can make methods on the API type so that the strings are consistent and can be referenced anywhere 

i.e.

```
func (a BuilderAPI) String() string {
    switch a {
      case BuilderAPIRefRecipient:
        return "refund-recipient"
      case BuilderAPIMevShareBeta1:
        return "v0.1"
      default: 
        return "unsupported"
    }
}

Reply:
I added `parseBuilderAPI` method to achieve a similar goals, I think it would be a bit more compact this way.